### ISSUE TITLE
don't know what I should add to support the `new` type

### ISSUE THREAD

**Issue by NormanPerrin** [eslint/eslint#10850]
### Question
The linter throws an error of type `new` unknow.
How could I config the eslint to support this type?

### type reference
[function-new-type](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#function-new-type)

### error
```
Syntax error in type: {new(source: {documents: Map<string, D>}): DListConstructor} jsdoc/valid-types
```

### .eslintrc config
```
{
    "plugins": [
        "jsdoc",
        "closure"
    ],
    "env": {
        "es6": true
    },
    "rules": {
        "jsdoc/check-param-names": 1,
        "jsdoc/check-tag-names": 1,
        "jsdoc/check-types": 1,
        "jsdoc/newline-after-description": 1,
        "jsdoc/require-description-complete-sentence": 1,
        "jsdoc/require-param": 1,
        "jsdoc/require-param-type": 1,
        "jsdoc/require-returns-type": 1,
        "jsdoc/valid-types": 1
    },
    "settings": {
        "jsdoc": {
            "additionalTagNames": {
                "customTags": ["template"]
            }
        }
    }
}
```

**Comment by platinumazure** [comment]
Hi @NormanPerrin, thanks for the issue.

This ESLint error is coming from [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc), not a core rule. So if you don't get a good answer here, you should open an issue on that repository.

That said, the declaration you supplied does look suspect:

```js
new({documents: Map<string,asasd D>}): DListConstructor
// Likely typo here: ------^
```

Maybe it should be something like this?

```js
new({documents: Map<string, D>}): DListConstructor
```

**Comment by kaicataldo** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
